src/validate_pyproject/formats.py

```diff
@@ -354,7 +356,9 @@ def python_entrypoint_reference(value: str) -> bool:
         obj = rest
 
     module_parts = module.split(".")
-    identifiers = _chain(module_parts, obj.split(".")) if rest else module_parts
+    identifiers: Iterable[str] = (
```

Comment:
Not sure I understand why Ruff is removing the `[]`, but it's also passing on Python <3.10, so maybe the get item was supported here earlier?

Reply:
I get TypeError: 'ABCMeta' object is not subscriptable when I try in on 3.18 by hand. Is there a future import annotations somewhere in this file?

Reply:
I wonder if we make the two sides of the `if...else` ternary operation have the same type, we would still need to explicitly annotate it. For example:

```diff
- identifiers: Iterable[str] = (
-     _chain(module_parts, obj.split(".")) if rest else module_parts
- )
+ identifiers = _chain(module_parts, obj.split(".")) if rest else iter(module_parts)
```

Reply:
I am just testing in the CI if that will work, otherwise we can revert the extra commit we pushed.

Reply:
It seems like it was okay in the past, so maybe it's a mypy issue, and it used to be the union? This is slightly more consistent anyway, though, as now both are iterators instead of one being an iterable.